src/modules/playback-engine/output/chromecast/chromecast-output.ts

```diff
@@ -0,0 +1,278 @@
+import {
+  CrossOriginValue,
+  ICromecastDebugInfo,
+  IPlayableSource,
+  IVideoOutput,
+  PlayableMediaSource,
+  PreloadType,
+} from '../../types';
+import RemotePlayerController = cast.framework.RemotePlayerController;
+import RemotePlayer = cast.framework.RemotePlayer;
+import { EngineState } from '../../../../constants';
+import StateEngine from './state-engine';
+import { IEventEmitter } from '../../../event-emitter/types';
+import logger from '../../../../utils/logger';
+import { getMimeByType, getMimeByUrl } from '../../../../utils/get-mime-type';
+
+const NOT_IMPLEMENTED = 'Not available using chromecast';
+
+export default class ChromecastOutput implements IVideoOutput {
+  type: string;
+
+  private _playerController: RemotePlayerController;
+  private _player: RemotePlayer;
+  private _stateEngine: StateEngine;
+  private _eventEmitter: IEventEmitter;
+  private _src: string;
+
+  private _isAutoplay: boolean;
+
+  constructor(eventEmitter: IEventEmitter) {
+    this._eventEmitter = eventEmitter;
+
+    this._initRemote();
+  }
+
+  play() {
+    if (this.isPaused) {
+      this._playerController.playOrPause();
+    }
+
+    return Promise.resolve();
+  }
+
+  pause() {
+    if (!this.isPaused) {
+      this._playerController.playOrPause();
+    }
+  }
+
+  setMute(mute: boolean) {
+    if (mute === undefined) {
+      return;
+    }
+
+    if (mute !== this.isMuted) {
+      this._playerController.muteOrUnmute();
+    }
+  }
+
+  setAutoplay(isAutoplay: boolean) {
+    this._isAutoplay = isAutoplay;
+  }
+
+  setInline() {
+    logger.warn(NOT_IMPLEMENTED);
+  }
+
+  setCrossOrigin() {
+    logger.warn(NOT_IMPLEMENTED);
+  }
+
+  setCurrentTime(time: number) {
+    this._player.currentTime = time;
+    this._playerController.seek();
+  }
+
+  setVolume(volume: number) {
+    if (this._player.volumeLevel === volume) {
+      return;
+    }
+
+    this._player.volumeLevel = volume;
+    this._playerController.setVolumeLevel();
+  }
+
+  setLoop() {
+    logger.warn(NOT_IMPLEMENTED);
+  }
+
+  setPlaybackRate() {
+    // TODO: https://developers.google.com/cast/docs/reference/chrome/chrome.cast.media.Media#playbackRate
+    // Works only via receiver API
+    logger.warn(NOT_IMPLEMENTED);
+  }
+
+  setPreload() {
+    logger.warn(NOT_IMPLEMENTED);
+  }
+
+  setSrc(source: PlayableMediaSource, startTime?: number) {
+    if (!source) {
+      return;
+    }
+    let _source = source;
```

Comment:
I don't think that it's good practice for naming to just add prefix `_`. Make it more logical, maybe `origin/final/resulted/etc` `Source`.